zulipterminal/ui_tools/boxes.py

```diff
@@ -118,31 +119,39 @@ def autocomplete_mentions(self, text: str, prefix_string: str
                               ) -> List[str]:
         # Handles user mentions (@ mentions and silent mentions)
         # and group mentions.
-        group_typeahead = ['@*{}*'.format(group_name)
-                           for group_name in self.model.user_group_names
-                           if match_group(group_name, text[1:])]
+        groups = [group_name
+                  for group_name in self.model.user_group_names
+                  if match_group(group_name, text[1:])]
+        group_typeahead = generate_typeaheads(groups, '@*{}*')
 
         users_list = self.view.users
-        user_typeahead = [prefix_string + '**{}**'.format(user['full_name'])
-                          for user in users_list
-                          if match_user(user, text[len(prefix_string):])]
+        users = [user['full_name']
+                 for user in users_list
+                 if match_user(user, text[len(prefix_string):])]
+        user_typeahead = generate_typeaheads(users, prefix_string + '**{}**')
+
         combined_typeahead = group_typeahead + user_typeahead
 
         return combined_typeahead
 
     def autocomplete_streams(self, text: str,
                              prefix_string: str) -> List[str]:
         streams_list = self.view.pinned_streams + self.view.unpinned_streams
-        stream_typeahead = [('#**{}**'.format(stream[0]), stream[0])
-                            for stream in streams_list]
-        return match_stream(stream_typeahead, text[1:],
+        streams = [stream[0]
+                   for stream in streams_list]
+        stream_typeahead = generate_typeaheads(streams, '#**{}**')
+        stream_data = list(zip(stream_typeahead, streams))
```

Comment:
This is a temp fix. @preetmishra what do you think about refactoring `match_streams` to match only using stream names? I don't mind doing it myself, asking for inputs.

Reply:
It could work for `autocomplete_streams` as the 'formatting' that is applied to each stream is simple. However, I think formatting/rebuilding buttons in `update_streams`, after every search,  would be an overhead.

Reply:
@neiljp I don't see where the nested tuples are being generated?

Reply:
My bad, found it. I had forgotten `generate_typeaheads` returns a list of tuples.